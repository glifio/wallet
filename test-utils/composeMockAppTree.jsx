/* eslint-disable react/prop-types */
import React, { useReducer } from 'react'
import cloneDeep from 'lodash.clonedeep'

import WalletProvider from '@openworklabs/filecoin-wallet-provider'
import { FilecoinNumber } from '@openworklabs/filecoin-number'
import { Provider } from 'react-redux'

import { NetworkCheck } from '../lib/check-network'
import BalancePoller from '../lib/update-balance'
import { ConverterContext } from '../lib/Converter'
import { theme, ThemeProvider } from '../components/Shared'
import { initializeStore } from './index'
import { initialState } from '../store/states'
import walletProviderReducer, {
  initialState as walletProviderInitialState,
  setWalletType,
  setError,
  resetLedgerState,
  resetState
} from '../WalletProvider/state'
import { WalletProviderContext } from '../WalletProvider'
import createPath from '../utils/createPath'
import { IMPORT_MNEMONIC } from '../constants'

const useRouter = jest.spyOn(require('next/router'), 'useRouter')

const mockRouter = jest.fn(() => {})
useRouter.mockImplementation(() => ({
  query: 'network=t',
  push: mockRouter,
  replace: mockRouter
}))

jest.mock('@openworklabs/filecoin-wallet-provider')
const mockGetAccounts = jest
  .fn()
  .mockImplementation(() =>
    Promise.resolve(['t1mbk7q6gm4rjlndfqw6f2vkfgqotres3fgicb2uq'])
  )

const mockGetBalance = jest
  .fn()
  .mockImplementation(() => Promise.resolve(new FilecoinNumber('1', 'fil')))

WalletProvider.mockImplementation(() => {
  return {
    wallet: {
      getAccounts: mockGetAccounts,
      sign: jest.fn().mockImplementation(() => 'xxxyyyyzzzz')
    },
    getBalance: mockGetBalance,
    getNonce: jest.fn().mockImplementation(() => 0),
    estimateGas: jest
      .fn()
      .mockImplementation(() => new FilecoinNumber('122', 'attofil')),
    sendMessage: jest.fn().mockImplementation(() => 'QmZCid!')
  }
})

const mockWalletProviderInstance = new WalletProvider()

const presets = {
  preOnboard: cloneDeep(initialState),
  postOnboard: cloneDeep({
    ...initialState,
    wallets: [
      {
        address: 't1z225tguggx4onbauimqvxzutopzdr2m4s6z6wgi',
        balance: new FilecoinNumber('1', 'fil'),
        path: createPath(1, 0)
      }
    ],
    selectedWalletIdx: 0
  })
}

const composeWalletProviderState = (initialWalletProviderState, preset) => {
  switch (preset) {
    case 'postOnboard': {
      return Object.freeze({
        ...initialWalletProviderState,
        walletType: IMPORT_MNEMONIC,
        walletProvider: mockWalletProviderInstance
      })
    }
    default:
      return initialWalletProviderState
  }
}

/**
 * VALID OPTIONS:
 * mockConverterInstance (an object that stubs the converter functionality)
 * walletProviderReducer (a reducer to stub the state handling of the wallet provider)
 * walletProviderInitialState(a reducer to stub the state of the walletProvider context)
 * pathname
 * query (usually { network: 'f' } or { network: 't' })
 */

export default (statePreset = 'preOnboard', options = {}) => {
  const state = options.state || presets[statePreset]
  const store = initializeStore(state)

  const mockConverterInstance = options.mockConverterInstance || {
    cacheConversionRate: jest.fn(),
    toFIL: jest.fn().mockImplementation(amount => {
      return new FilecoinNumber(amount, 'fil').dividedBy(5)
    }),
    fromFIL: jest.fn().mockImplementation(amount => {
      return new FilecoinNumber(amount, 'fil').multipliedBy(5)
    })
  }

  const pathname = options.pathname || '/wallet'
  const query = options.query || { network: 't' }

  const Tree = ({ children }) => {
    const [initialWalletProviderState, walletProviderDispatch] = useReducer(
      options.reducer || walletProviderReducer,
      options.walletProviderInitialState || walletProviderInitialState
    )

    const walletProviderState = composeWalletProviderState(
      initialWalletProviderState,
      statePreset
    )
    return (
      <Provider store={store}>
        <ConverterContext.Provider
          value={{
            converter: mockConverterInstance,
            converterError: options.converterError || null
          }}
        >
          <WalletProviderContext.Provider
            value={{
              state: walletProviderState,
              dispatch: walletProviderDispatch,
              fetchDefaultWallet: jest.fn().mockImplementation(),
              setWalletError: jest
                .fn()
                .mockImplementation(errorMessage =>
                  walletProviderDispatch(setError(errorMessage))
                ),
              setWalletType: jest
                .fn()
                .mockImplementation(walletType =>
                  walletProviderDispatch(setWalletType(walletType))
                ),
              setLedgerProvider: jest.fn(),
              connectLedger: jest.fn(),
              resetLedgerState: jest.fn().mockImplementation(() => {
                walletProviderDispatch(resetLedgerState())
              }),
              resetState: jest.fn().mockImplementation(() => {
                walletProviderDispatch(resetState())
              })
            }}
          >
            <NetworkCheck
              networkFromRdx={store.getState().network}
              pathname={pathname}
              query={query}
              switchNetwork={jest.fn()}
            />
            <BalancePoller />
            <ThemeProvider theme={theme}>{children}</ThemeProvider>
          </WalletProviderContext.Provider>
        </ConverterContext.Provider>
      </Provider>
    )
  }

  return { Tree, store, walletProvider: mockWalletProviderInstance }
}
