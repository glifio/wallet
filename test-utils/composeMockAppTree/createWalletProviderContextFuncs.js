import { FilecoinNumber } from '@glif/filecoin-number'

import {
  setWalletType,
  setError,
  resetLedgerState,
  resetState
} from '../../WalletProvider/state'

export default walletProviderDispatch => ({
  fetchDefaultWallet: jest.fn().mockImplementation(() => ({
    balance: new FilecoinNumber('1', 'fil'),
    address: 't1mbk7q6gm4rjlndfqw6f2vkfgqotres3fgicb2uq',
    path: "m/44'/1'/0/0/0"
  })),
  setWalletError: jest
    .fn()
    .mockImplementation(errorMessage =>
      walletProviderDispatch(setError(errorMessage))
    ),
  setWalletType: jest
    .fn()
    .mockImplementation(walletType =>
      walletProviderDispatch(setWalletType(walletType))
    ),
  setLedgerProvider: jest.fn(),
  connectLedger: jest.fn(),
  resetLedgerState: jest.fn().mockImplementation(() => {
    walletProviderDispatch(resetLedgerState())
  }),
  resetState: jest.fn().mockImplementation(() => {
    walletProviderDispatch(resetState())
  })
})
