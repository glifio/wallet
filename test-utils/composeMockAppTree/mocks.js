import WalletProvider from '@openworklabs/filecoin-wallet-provider'
import { FilecoinNumber } from '@openworklabs/filecoin-number'

import {
  setWalletType,
  setError,
  resetLedgerState,
  resetState
} from '../../WalletProvider/state'
import { SINGLE_KEY, HD_WALLET, LEDGER } from '../../constants'

jest.mock('@openworklabs/filecoin-wallet-provider')

const useRouter = jest.spyOn(require('next/router'), 'useRouter')

export const mockRouterReplace = jest.fn(() => {})
export const mockRouterPush = jest.fn(() => {})
const unix = jest.spyOn(require('dayjs'), 'unix')

unix.mockImplementation(() => {
  return {
    format: fmt => {
      if (fmt === 'MMM DD') return 'Jan 12'
      if (fmt === 'hh:mmA') return '2:30PM'
    }
  }
})

useRouter.mockImplementation(() => ({
  query: 'network=t',
  push: mockRouterPush,
  replace: mockRouterReplace
}))

const mockGetAccounts = jest
  .fn()
  .mockImplementation(() =>
    Promise.resolve(['t1mbk7q6gm4rjlndfqw6f2vkfgqotres3fgicb2uq'])
  )

const mockGetBalance = jest
  .fn()
  .mockImplementation(() => Promise.resolve(new FilecoinNumber('1', 'fil')))

WalletProvider.mockImplementation(() => {
  return {
    wallet: {
      getAccounts: mockGetAccounts,
      sign: jest.fn().mockImplementation(() => 'xxxyyyyzzzz')
    },
    getBalance: mockGetBalance,
    getNonce: jest.fn().mockImplementation(() => 0),
    estimateGas: jest
      .fn()
      .mockImplementation(() => new FilecoinNumber('122', 'attofil')),
    sendMessage: jest.fn().mockImplementation(() => 'QmZCid!')
  }
})

export const mockWalletProviderInstance = new WalletProvider()

export const defaultMockConverterInstance = {
  cacheConversionRate: jest.fn(),
  toFIL: jest.fn().mockImplementation(amount => {
    return new FilecoinNumber(amount, 'fil').dividedBy(5)
  }),
  fromFIL: jest.fn().mockImplementation(amount => {
    return new FilecoinNumber(amount, 'fil').multipliedBy(5)
  })
}

export const createMockWalletProviderContextFuncs = walletProviderDispatch => ({
  fetchDefaultWallet: jest.fn().mockImplementation(() => ({
    balance: new FilecoinNumber('1', 'fil'),
    address: 't1mbk7q6gm4rjlndfqw6f2vkfgqotres3fgicb2uq',
    path: ''
  })),
  setWalletError: jest
    .fn()
    .mockImplementation(errorMessage =>
      walletProviderDispatch(setError(errorMessage))
    ),
  setWalletType: jest
    .fn()
    .mockImplementation(walletType =>
      walletProviderDispatch(setWalletType(walletType))
    ),
  setLedgerProvider: jest.fn(),
  connectLedger: jest.fn(),
  resetLedgerState: jest.fn().mockImplementation(() => {
    walletProviderDispatch(resetLedgerState())
  }),
  resetState: jest.fn().mockImplementation(() => {
    walletProviderDispatch(resetState())
  })
})

class MockSingleKeyProvider {
  type = SINGLE_KEY

  getAccounts = () => jest.fn()

  sign = () => jest.fn()
}

class MockHDWalletProvider {
  type = HD_WALLET

  getAccounts = () => jest.fn()

  sign = () => jest.fn()
}

export const mockWalletSubproviders = {
  SingleKeyProvider: MockSingleKeyProvider,
  HDWalletProvider: MockHDWalletProvider
}
