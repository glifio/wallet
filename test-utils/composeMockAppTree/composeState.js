import cloneDeep from 'lodash.clonedeep'
import { FilecoinNumber } from '@openworklabs/filecoin-number'

import createPath from '../../utils/createPath'
import { IMPORT_MNEMONIC } from '../../constants'
import { initialState } from '../../store/states'
import { mockWalletProviderInstance } from '../mocks/mock-wallet-provider'

export const presets = {
  preOnboard: cloneDeep(initialState),
  postOnboard: cloneDeep({
    ...initialState,
    wallets: [
      {
        address: 't1z225tguggx4onbauimqvxzutopzdr2m4s6z6wgi',
        balance: new FilecoinNumber('1', 'fil'),
        path: createPath(1, 0)
      }
    ],
    selectedWalletIdx: 0
  }),
  postOnboardWithError: cloneDeep({
    ...initialState,
    error: 'error for testing',
    wallets: [
      {
        address: 't1z225tguggx4onbauimqvxzutopzdr2m4s6z6wgi',
        balance: new FilecoinNumber('1', 'fil'),
        path: createPath(1, 0)
      }
    ],
    selectedWalletIdx: 0
  })
}

export const composeWalletProviderState = (
  initialWalletProviderState,
  preset
) => {
  switch (preset) {
    case 'postOnboard': {
      return Object.freeze({
        ...initialWalletProviderState,
        walletType: IMPORT_MNEMONIC,
        walletProvider: mockWalletProviderInstance
      })
    }
    case 'postOnboardWithError': {
      return Object.freeze({
        ...initialWalletProviderState,
        walletType: IMPORT_MNEMONIC,
        walletProvider: mockWalletProviderInstance
      })
    }
    default:
      return initialWalletProviderState
  }
}
