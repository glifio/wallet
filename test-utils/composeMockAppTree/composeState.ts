import cloneDeep from 'lodash.clonedeep'
import { FilecoinNumber } from '@glif/filecoin-number'

import createPath from '../../utils/createPath'
import { IMPORT_MNEMONIC } from '../../constants'
import { initialState } from '../../store/states'
import { mockWalletProviderInstance } from '../mocks/mock-wallet-provider'
import { emptyMsigState } from '../../MsigProvider/types'
import { WALLET_ADDRESS, MULTISIG_ACTOR_ADDRESS, signers } from '../constants'

export const presets = {
  preOnboard: cloneDeep(initialState),
  postOnboard: cloneDeep({
    ...initialState,
    wallets: [
      {
        address: WALLET_ADDRESS,
        balance: new FilecoinNumber('1', 'fil'),
        path: createPath(1, 0)
      }
    ],
    selectedWalletIdx: 0
  }),
  postOnboardLowBal: cloneDeep({
    ...initialState,
    wallets: [
      {
        address: WALLET_ADDRESS,
        balance: new FilecoinNumber('.000001', 'fil'),
        path: createPath(1, 0)
      }
    ],
    selectedWalletIdx: 0
  }),
  postOnboardWithError: cloneDeep({
    ...initialState,
    error: 'error for testing',
    wallets: [
      {
        address: WALLET_ADDRESS,
        balance: new FilecoinNumber('1', 'fil'),
        path: createPath(1, 0)
      }
    ],
    selectedWalletIdx: 0
  }),
  pendingMsigCreate: cloneDeep({
    ...initialState,
    wallets: [
      {
        address: WALLET_ADDRESS,
        balance: new FilecoinNumber('1', 'fil'),
        path: createPath(1, 0)
      }
    ],
    selectedWalletIdx: 0,
    messages: {
      confirmed: [],
      pending: [
        {
          cid: 'bafy2bzaced2godar6dht6ag6omfa53jr6tykygtzskzodqo2k4ngvyou4yct2',
          from: 'f1nq5k2mps5umtebdovlyo7y6a3ywc7u4tobtuo3a',
          gasFeeCap: '100425',
          gasLimit: 11452096,
          gasPremium: '99371',
          maxFee: '1150076740800',
          method: 'EXEC',
          nonce: 56,
          paidFee: '0',
          params: {
            num_approvals_threshold: 1,
            signers: ['f1nq5k2mps5umtebdovlyo7y6a3ywc7u4tobtuo3a'],
            start_epoch: '0',
            unlock_duration: '0'
          },
          timestamp: 1632317830,
          to: 'f01'
        }
      ]
    }
  })
}

export const composeWalletProviderState = initialWalletProviderState => {
  return Object.freeze({
    ...initialWalletProviderState,
    walletType: IMPORT_MNEMONIC,
    walletProvider: mockWalletProviderInstance
  })
}

export const mockMsigProviderContext = {
  Address: MULTISIG_ACTOR_ADDRESS,
  ActorCode: 'fil/5/multisig',
  Balance: new FilecoinNumber('1', 'fil'),
  AvailableBalance: new FilecoinNumber('1', 'fil'),
  Signers: signers,
  InitialBalance: new FilecoinNumber('1', 'fil'),
  NextTxnID: 0,
  NumApprovalsThreshold: 1,
  StartEpoch: 0,
  UnlockDuration: 0,
  errors: {
    notMsigActor: false,
    connectedWalletNotMsigSigner: false,
    actorNotFound: false,
    unhandledError: ''
  },
  loading: false,
  setMsigActor: null
}

export const composeMsigProviderState = (preset: keyof typeof presets) => {
  switch (preset) {
    case 'preOnboard': {
      return Object.freeze(emptyMsigState)
    }
    case 'pendingMsigCreate': {
      return Object.freeze(emptyMsigState)
    }
    default: {
      return Object.freeze(mockMsigProviderContext)
    }
  }
}
