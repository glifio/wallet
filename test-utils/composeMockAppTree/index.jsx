/* eslint-disable react/prop-types */
import React, { useReducer } from 'react'
import { Provider } from 'react-redux'
import { Converter } from '@openworklabs/filecoin-number'

import { NetworkCheck } from '../../lib/check-network'
import BalancePoller from '../../lib/update-balance'
import { ConverterContext } from '../../lib/Converter'
import { theme, ThemeProvider } from '../../components/Shared'
import { initializeStore } from '..'
import walletProviderReducer, {
  initialState as walletProviderInitialState
} from '../../WalletProvider/state'
import { WasmContext } from '../../lib/WasmLoader'
import { WalletProviderContext } from '../../WalletProvider'
import { mockWalletProviderInstance } from '../mocks/mock-wallet-provider'
import createMockWalletProviderContextFuncs from './createWalletProviderContextFuncs.js'
import mockWalletSubproviders from '../mocks/mock-wallet-subproviders'
import { presets, composeWalletProviderState } from './composeState'

/**
 * VALID OPTIONS:
 * mockConverterInstance (an object that stubs the converter functionality)
 * walletProviderReducer (a reducer to stub the state handling of the wallet provider)
 * walletProviderInitialState(a reducer to stub the state of the walletProvider context)
 * walletProviderDispatch(a dispatcher to mock/read calls to the walletProvider's state)
 * pathname
 * query (usually { network: 'f' } or { network: 't' })
 */

export default (statePreset = 'preOnboard', options = {}) => {
  const state = options.state || presets[statePreset]
  const store = initializeStore(state)

  const pathname = options.pathname || '/wallet'
  const query = options.query || { network: 't' }

  const Tree = ({ children }) => {
    const [initialWalletProviderState, walletProviderDispatch] = useReducer(
      options.reducer || walletProviderReducer,
      options.walletProviderInitialState || walletProviderInitialState
    )

    const walletProviderState = composeWalletProviderState(
      initialWalletProviderState,
      statePreset
    )

    const mockWalletProviderContextFuncs = createMockWalletProviderContextFuncs(
      options.walletProviderDispatch || walletProviderDispatch
    )

    return (
      <Provider store={store}>
        <WasmContext.Provider
          value={{
            generateMnemonic: jest
              .fn()
              .mockImplementation(
                () =>
                  'slender spread awkward chicken noise useful thank dentist tip bronze ritual explain version spot collect whisper glow peanut bus local country album punch frown'
              )
          }}
        >
          <ConverterContext.Provider
            value={{
              converter: new Converter(),
              converterError: options.converterError || null
            }}
          >
            <WalletProviderContext.Provider
              value={{
                state: walletProviderState,
                dispatch:
                  options.walletProviderDispatch || walletProviderDispatch,
                walletSubproviders: {
                  ...mockWalletSubproviders
                },
                ...mockWalletProviderContextFuncs
              }}
            >
              <NetworkCheck
                networkFromRdx={store.getState().network}
                pathname={pathname}
                query={query}
                switchNetwork={jest.fn()}
              />
              <BalancePoller />
              <ThemeProvider theme={theme}>{children}</ThemeProvider>
            </WalletProviderContext.Provider>
          </ConverterContext.Provider>
        </WasmContext.Provider>
      </Provider>
    )
  }

  return {
    Tree,
    store,
    walletProvider: mockWalletProviderInstance
  }
}
