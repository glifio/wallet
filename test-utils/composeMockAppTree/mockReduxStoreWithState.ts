import { presets } from './composeState'
import { initializeStore } from '..'

export default function mockReduxStoreWithState(options: {
  state?: object
  statePreset?: string
}) {
  const state = options.state || presets[options.statePreset]
  return initializeStore(state)
}
