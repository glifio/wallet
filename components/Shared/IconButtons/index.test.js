import { cleanup, render, act, fireEvent } from '@testing-library/react'
import { ButtonClose, ButtonCopyAccountAddress } from '.'
import composeMockAppTree from '../../../test-utils/composeMockAppTree'

describe('ButtonClose', () => {
  afterEach(cleanup)
  test('renders the ButtonClose', () => {
    const { Tree } = composeMockAppTree('postOnboard')
    const { container } = render(
      <Tree>
        <ButtonClose onClick={() => {}} />
      </Tree>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('clicking ButtonClose calls onClick', () => {
    const { Tree } = composeMockAppTree('postOnboard')
    const mockOnClick = jest.fn()
    const { getByTitle } = render(
      <Tree>
        <ButtonClose title='Close' disabled={false} onClick={mockOnClick} />
      </Tree>
    )

    act(() => {
      fireEvent.click(getByTitle('Close'))
    })

    expect(mockOnClick).toHaveBeenCalled()
  })
})

describe('ButtonCopyAccountAddress', () => {
  afterEach(cleanup)
  test('renders the ButtonCopyAccountAddress', () => {
    const { Tree } = composeMockAppTree('postOnboard')
    const { container } = render(
      <Tree>
        <ButtonCopyAccountAddress onClick={() => {}} />
      </Tree>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('clicking ButtonCopyAccountAddress calls onClick', () => {
    const { Tree } = composeMockAppTree('postOnboard')
    const mockOnClick = jest.fn()
    const { getByTitle } = render(
      <Tree>
        <ButtonCopyAccountAddress title='Copy' onClick={mockOnClick} />
      </Tree>
    )

    act(() => {
      fireEvent.click(getByTitle('Copy'))
    })

    expect(mockOnClick).toHaveBeenCalled()
  })
})
