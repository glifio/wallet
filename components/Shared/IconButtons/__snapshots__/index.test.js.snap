// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonClose renders the ButtonClose 1`] = `
.c1 {
  width: 24;
  height: 24;
}

.c0 {
  outline: 0;
  border: 0;
  background: transparent;
  -webkit-transition: 0.24s ease-in-out;
  transition: 0.24s ease-in-out;
  cursor: pointer;
  display: inline-block;
}

.c0:hover {
  -webkit-transform: scale(1.25);
  -ms-transform: scale(1.25);
  transform: scale(1.25);
}

<button
  class="c0"
  display="inline-block"
>
  <svg
    class="c1"
    fill="none"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M19.7333 4.2669C19.3776 3.91103 18.801 3.91103 18.4453 4.2669L11.9994 10.7166L5.55552 4.26885C5.19986 3.91299 4.62323 3.91299 4.26757 4.26885C3.91191 4.62472 3.91191 5.20169 4.26757 5.55756L10.7115 12.0053L4.27793 18.4426C3.92228 18.7985 3.92228 19.3755 4.27793 19.7313C4.63359 20.0872 5.21022 20.0872 5.56587 19.7313L11.9994 13.294L18.435 19.7333C18.7906 20.0892 19.3672 20.0892 19.7229 19.7333C20.0786 19.3774 20.0786 18.8005 19.7229 18.4446L13.2874 12.0053L19.7333 5.5556C20.0889 5.19974 20.0889 4.62276 19.7333 4.2669Z"
      fill="#5E26FF"
      fill-rule="evenodd"
    />
  </svg>
</button>
`;

exports[`ButtonCopyAccountAddress renders the ButtonCopyAccountAddress 1`] = `
.c1 {
  width: 24;
  height: 24;
}

.c0 {
  outline: 0;
  border: 0;
  background: transparent;
  -webkit-transition: 0.24s ease-in-out;
  transition: 0.24s ease-in-out;
  cursor: pointer;
  display: inline-block;
}

.c0:hover {
  -webkit-transform: scale(1.25);
  -ms-transform: scale(1.25);
  transform: scale(1.25);
}

<button
  class="c0"
  display="inline-block"
>
  <svg
    class="c1"
    fill="none"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="#1A0066"
      height="12"
      width="12"
      x="8"
      y="4"
    />
    <rect
      fill="#E0D7FE"
      height="12"
      width="12"
      x="4"
      y="8"
    />
  </svg>
</button>
`;
